File: src/main/kotlin/org/kite9/diagram/visualization/planarization/rhd/position/PositionRoutingInfo.java
package org.kite9.diagram.visualization.planarization.rhd.position;

import java.text.DecimalFormat;
import java.text.NumberFormat;

import org.kite9.diagram.common.elements.RoutingInfo;

public abstract class PositionRoutingInfo implements RoutingInfo {
	
	public static final NumberFormat nf = new DecimalFormat(".0000");

	public abstract double getMinX();
	public abstract double getMaxX();
	public abstract double getMinY();
	public abstract double getMaxY();
	public abstract double getWidth();
	public abstract double getHeight();
	public abstract boolean isBreakingOrder();
	
	@Override
	public String outputX() {
		return nf.format(getMinX())+"-"+nf.format(getMaxX());
	}
	
	@Override
	public String outputY() {
		return nf.format(getMinY())+"-"+nf.format(getMaxY());
	}
	
	public String toString() {
		return outputX() + ", "+ outputY();
	}
	
}

File: src/main/kotlin/org/kite9/diagram/visualization/planarization/rhd/position/PositionRoutingInfo.kt
package org.kite9.diagram.visualization.planarization.rhd.position

import org.kite9.diagram.common.elements.RoutingInfo

abstract class PositionRoutingInfo : RoutingInfo {

    abstract val minX: Double
    abstract val maxX: Double
    abstract val minY: Double
    abstract val maxY: Double
    abstract val width: Double
    abstract val height: Double
    abstract val isBreakingOrder: Boolean

    override fun outputX(): String {
        return "${minX}-${maxX}"
    }

    override fun outputY(): String {
        return "${minY}-${maxY}"
    }

    override fun toString(): String {
        return outputX() + ", " + outputY()
    }

}